client/src/Components/CarData.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { useQuery } from '@apollo/react-hooks';
+import { getDashboardAllDataQuery } from '../queries/queries';
+import { Card } from 'semantic-ui-react';
+import LoadingSpinner from './LoadingSpinner';
+import LastFillUpCard from './Card/LastFillUpCard';
+import FuelLeftCard from './Card/FuelLeftCard';
+import DiagnosticCard from './Card/DiagnosticCard';
+import BusinessRatioCard from './Card/BusinessRatioCard';
+import AverageSpeedCard from './Card/AverageSpeedCard';
+import TravelDistanceTotalCard from './Card/TravelDistanceTotalCard';
+import TimeInCarCard from './Card/TimeInCarCard';
+import EmissionsCard from './Card/EmissionsCard';
+import FuelEconomyCard from './Card/FuelEconomyCard';
+import GoogleMap from './GoogleMap/GoogleMap';
+import AntdTableComponent from './AntdTable/AntdTableComponent';
+
+export default function CarData(props) {
+  const { loading, error, data } = useQuery(getDashboardAllDataQuery, {
+    variables: {
+      vehicleId: props.vehicleIdState.value,
```

Comment:
nitpick `props.vehicleId` would have been fine. It's confusing using the word state for something that is a prop not state.